src/analysis/retail/rogue/subtlety/modules/spells/Flagellation.tsx

```diff
@@ -135,72 +135,109 @@ export default class FlagellationAnalysis extends Analyzer {
     hasSymbolsAvailable: boolean,
     hasShadowDanceAvailable: boolean,
     hasSecretTechniqueAvailable: boolean,
+  ): ChecklistUsageInfo[] {
+    return [
+      this.bladesPerformance(event, hasBladesAvailable),
+      this.symbolsPerformance(event, hasSymbolsAvailable),
+      this.shadowDancePerformance(event, hasShadowDanceAvailable),
+      this.secretTechniquePerformance(event, hasSecretTechniqueAvailable),
+    ].filter((performance): performance is ChecklistUsageInfo => performance !== undefined);
+  }
+
+  private bladesPerformance(
+    event: CastEvent,
+    hasBladesAvailable: boolean,
+  ): ChecklistUsageInfo | undefined {
+    return createChecklistItem(
+      'flagellation_blades',
+      { event },
+      {
+        performance: hasBladesAvailable ? QualitativePerformance.Good : QualitativePerformance.Fail,
+        summary: <div>Shadow Blades Cooldown</div>,
```

Comment:
for future ref, you can use react fragments instead of divs here (and in similar locations). don't need to change it at this point though

```suggestion
        summary: <>Shadow Blades Cooldown</>,
```